cli-toolchain/usage.md

```diff
@@ -419,6 +419,236 @@ Now, when Dev Corp is trying to initialize an Amplify Project, the user can sele
+
+If you do not wish to test your app locally, you can still use the [local GraphQL console](#graphql-local-console) as well as [edit, debug, and test your VTL resolvers](#graphql-resolver-debugging) locally against the mock endpoint.
+
+When adding a schema use an API Key at first to ensure everything works, though you can authenticate against a Cognito User Pool and the local testing server will honor the JWT tokens. You can also mock the JWT tokens in the local console (outlined below), however in that case you will need to do an `amplify push` first to create the User Pool.
+
+When defining a schema you can use directives from the GraphQL Transformer in local testing as well as local code generation from the schema for types. The following directives are currently supported in local testing:
+
+- [@auth](./graphql#auth)
+- [@key](./graphql#key)
+- [@connection](./graphql#connection)
+- [@versioned](./graphql#versioned)
+- [@function](./graphql#function)
+
+Note that `@searchable` will not be supported for initial release.
+
+### Storage mocking setup
+For S3 storage mocking, after running `amplify init` you must first run through `amplify add auth`, either explicitly or implicitly if adding storage first, and then run an `amplify push`. This is because mocking storage in client libraries requires credentials for initial setup. Note however that S3 authorization rules, such as those placed on a bucket policy, are not checked by local mocking at this time. 
+
+Once you have done an initial push you can run the mock server and hit the local endpoint:
+
+```terminal
+$ amplify init
+$ amplify add storage #This will prompt you to add auth
+$ amplify push
+$ amplify mock storage
+```
+
+To use an iOS application with the local S3 endpoint you will need to [modify your Info.plist file](#ios-config). To use an Android application with the local S3 endpoint you will need an extra [configuration in your AndroidManifest.xml](#android-config).
+
+For DynamoDB storage, setup is automatically done when creating a GraphQL API and no action is needed on your part.
+
+
+### Function mocking setup
+For Lambda function mocking, after running `amplify init` you can add a function to your project with `amplify add function` and either mock invoke it directly, or use the [@function](./graphql#function) directive as part of your GraphQL schema to mock the invocation as part of your API.
+
+To invoke the function with the local mock invoke:
+
+```terminal
+$ amplify init
+$ amplify add function ## Follow prompts
+$ amplify mock function <function_name>
+```
+
+This will take you through a few questions, such as the entry point for your Lambda function and sample event context to pass. The defaults are `index.js` and `event.json`.
+
+Note that you will need to run `yarn` or `npm install` first if your Lambda function has any external dependencies. Only Node.js functions are supported at this time.
```

Comment:
I would mention where to go on the terminal before running `yarn` or `npm install`.

```suggestion
Note that you will need to run `yarn` or `npm install` first if your Lambda function has any external dependencies (Go to `<project-dir>/amplify/backend/function/<function-name>/src`). Only Node.js functions are supported at this time.
```